sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -158,7 +162,9 @@ case class Like(left: Expression, right: Expression, escapeChar: Char)
         val regexStr =
           StringEscapeUtils.escapeJava(escape(rVal.asInstanceOf[UTF8String].toString()))
         val pattern = ctx.addMutableState(patternClass, "patternLike",
-          v => s"""$v = $patternClass.compile("$regexStr");""")
+          v =>
+            s"""$v = $patternClass.compile("$regexStr",
+               |CollationSupport.collationAwareRegexFlags($collationId));""".stripMargin)
```

Comment:
`collationId` is static here. We should not generate java code to invoke `collationAwareRegexFlags`, but simply generate java int literal using the result of `collationAwareRegexFlags`

Reply:
that would be best - done in recent commit